backend/amd64/emit.mlp

```diff
@@ -718,18 +718,24 @@ let emit_global_label s =
 let move (src : Reg.t) (dst : Reg.t) =
   if src.loc <> dst.loc then
     begin match src.typ, src.loc, dst.typ, dst.loc with
-    | Float, Reg.Reg _, Float, Reg.Reg _
-    | Vec128, Reg.Reg _, Vec128, Reg.Reg _ -> I.movapd (reg src) (reg dst)
+    | (Float | Vec128), Reg.Reg _, (Float | Vec128), Reg.Reg _ ->
+      I.movapd (reg src) (reg dst)
     | Vec128, _, Vec128, _ ->
       I.movupd (reg src) (reg dst)
-    | Float, _, Float, _ -> I.movsd (reg src) (reg dst)
-    | Float, _, Int, _
-    | Int, _, Float, _ -> I.movq (reg src) (reg dst)
+    | Float, _, Float, _ ->
+      I.movsd (reg src) (reg dst)
+    | Float, _, Int, _ | Int, _, Float, _ ->
+      I.movq (reg src) (reg dst)
```

Comment:
Please can you add a CR here while you are at it about bits_of_float / float_of_bits and the plan to get rid of this case soon, using a separate `Ispecific`.